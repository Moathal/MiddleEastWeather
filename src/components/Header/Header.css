.header-container {
  display: flex;
  align-items: center;
  width: 100%;
  border: none;
  margin-bottom: 12px;
  background-color: rgb(0, 69, 134);
  box-shadow: 0 0 6px rgb(0, 29, 56);
  color: lightgray;
}

.website-identifier {
  display: flex;
  align-items: center;
  gap: 5px;
  margin: 12px;
  justify-self: flex-end;
}

.backIcon {
  height: 20px;
  margin: 5px;
  margin-left: 10px;
  justify-self: flex-start;
}

.hidden {
  display: none;
}

.logo {
  width: 40px;
  height: 40px;
}
