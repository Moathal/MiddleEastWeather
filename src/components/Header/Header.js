import React from 'react';
import { useLocation, Link } from 'react-router-dom';
import favicon from '../../assets/favicon.png';
import back from '../../assets/backIcon.ico.png';
import './Header.css';

function Header() {
  const location = useLocation();
  return (
    <div className="header-container">
      <Link className={location.pathname === '/' ? 'hidden' : 'show'} to="/" current={location.pathname}>
        <img className="backIcon" src={back} alt="back arrow" />
      </Link>
      <div className="website-identifier">
        <img className="logo" src={favicon} alt="logo" />
        <h1>MEWeather</h1>
      </div>
    </div>
  );
}

export default Header;
