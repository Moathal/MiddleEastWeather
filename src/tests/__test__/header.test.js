import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import Header from '../../components/Header/Header';

it('renders Header correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <Header />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
