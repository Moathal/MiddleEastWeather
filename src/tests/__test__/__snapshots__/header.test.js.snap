// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header correctly 1`] = `
<div
  className="header-container"
>
  <a
    className="hidden"
    current="/"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="back arrow"
      className="backIcon"
      src="backIcon.ico.png"
    />
  </a>
  <div
    className="website-identifier"
  >
    <img
      alt="logo"
      className="logo"
      src="favicon.png"
    />
    <h1>
      MEWeather
    </h1>
  </div>
</div>
`;
